Content.Server/Objectives/Systems/StealConditionSystem.cs

```diff
@@ -44,11 +47,17 @@ public override void Initialize()
     /// start checks of target acceptability, and generation of start values.
     private void OnAssigned(Entity<StealConditionComponent> condition, ref ObjectiveAssignedEvent args)
     {
+        var selectedMapId = _gameTicker.DefaultMap;
         List<StealTargetComponent?> targetList = new();
 
         var query = AllEntityQuery<StealTargetComponent>();
```

Comment:
```suggestion
        var query = AllEntityQuery<StealTargetComponent, TransformComponent>();
```
just get the component as part of the query